cmd/nerdctl/run_network_linux_test.go

```diff
@@ -529,3 +529,53 @@ func TestSharedNetworkStack(t *testing.T) {
 	base.Cmd("exec", containerNameJoin, "wget", "-qO-", "http://127.0.0.1:80").
 		AssertOutContains(testutil.NginxAlpineIndexHTMLSnippet)
 }
+
+func TestRunContainerWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		Ok      bool
+	}{
+		{"host", false},
+		{"none", true},
+		{"container:whatever" + tID, false},
+		{"bridge", true},
+		{networkBridge, true},
+		{networkMACvlan, true},
+		{networkIPvlan, false},
+	}
+	for _, test := range tests {
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		cmd := base.Cmd("run", "--rm", "--network", test.Network, "--mac-address", macAddress, testutil.CommonImage, "ip", "link", "show")
```

Comment:
Another way is to check the mac address by `cat sys/class/net/eth0/address`. If the network is `none`, it will return error.

Reply:
That's great!